src/ServicePulse.Host/vue/src/components/failedmessages/OrderBy.vue

```diff
@@ -1,14 +1,22 @@
-<script setup>
+<script setup lang="ts">
 import { onMounted, ref } from "vue";
 import { useCookies } from "vue3-cookies";
-
-const emit = defineEmits(["sortUpdated"]);
-
-const props = defineProps({
-  hideSort: Boolean,
-  sortOptions: Array,
-  sortSavePrefix: String,
-});
+import SortOptions, { GroupType } from "@/resources/SortOptions";
+
+const emit = defineEmits<{
+  sortUpdated: [option: SortOptions];
+}>();
+
+const props = withDefaults(
+  defineProps<{
+    hideSort?: boolean;
```

Comment:
is there any reason to mark this as nullable when it has a default?

Reply:
I believe it has to be nullable so that the caller does not have to include it, but giving it a default allows the dev tools to recognize that it will not be null.